### PR TITLE
docs & config

### PR DESCRIPTION
**Pull request by Okm165**

### CODE REVIEW

**Comment by Okm165** [comment]
@tdelabro let me know if u would add anything here

File: crates/air/build.rs [hunk 1]
```diff
@@ -21,5 +21,45 @@ fn main() {
...
+        #[rustfmt::skip]
+        mod check_layout {
+            check_feature_enabled!("dex", "recursive", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("dex", "recursive", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("recursive", "dex", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("recursive_with_poseidon", "dex", "recursive", "small", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("small", "dex", "recursive", "recursive_with_poseidon", "starknet", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("starknet", "dex", "recursive", "recursive_with_poseidon", "small", "starknet_with_keccak", "dynamic");
+            check_feature_conflict!("starknet_with_keccak", "dex", "recursive", "recursive_with_poseidon", "small", "starknet", "dynamic");
+            check_feature_conflict!("dynamic", "dex", "recursive", "recursive_with_poseidon", "small", "starknet", "starknet_with_keccak");
```
Comment (tdelabro) [inline crates/air/build.rs:55]: Pretty verbose and hard to maintain. But maybe there is no other way to express this.

File: crates/pow/build.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                ));
+            )*
+        };
+    }
+
+    #[rustfmt::skip]
+    mod check_hash {
+        check_feature_enabled!("keccak", "blake2s");
+        check_feature_conflict!("keccak", "blake2s");
+        check_feature_conflict!("blake2s", "keccak");
```
Comment (tdelabro) [inline crates/pow/build.rs:27]: I think it will never be triggered. The previous line will catch it

File: crates/air/build.rs (hunk 1 above)
Comment (Okm165) [reply crates/air/build.rs:55]: There is much better way actually

**Pull request merged**